.github/workflows/naming-convention.yml

```diff
@@ -4,48 +4,23 @@ on:
   push:
     branches: ["main"]
   pull_request:
-    # The branches below must be a subset of the branches above
     branches: ["main"]
   schedule:
     - cron: "43 22 * * 3"
-
 jobs:
-  commitlint:
-    name: Lint Commit Messages
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-
-      - name: Install Commitlint
-        run: |
-          npm install --save-dev @commitlint/{config-conventional,cli}
-
-      - name: Lint Commit Messages
-        run: |
-          if [ "${{ github.event_name }}" = "pull_request" ]; then
-            npx commitlint --from=${{ github.event.pull_request.head.sha }}
-          else
-            npx commitlint --from=${{ github.sha }}
-          fi
-
   eslint:
     name: Run eslint scanning
-    needs: commitlint
     runs-on: ubuntu-latest
     permissions:
       contents: read
       security-events: write
-      actions: read # only required for a private repository by github/codeql-action/upload-sarif to get the Action run status
+      actions: read
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
 
-      - name: Install ESLint
-        run: |
-          npm install eslint@8.10.0
-          npm install @microsoft/eslint-formatter-sarif@2.1.7
+      - name: Install dependencies
+        run: npm install
+
       - name: Run ESLint
-        run: npx eslint . --ext .ts,.tsx
+        run: npx eslint . --ext .ts,.tsx --fix
```

Comment:
You shouldn't run `--fix` in the CI. Else, it'll change the files in the CI, and will always be a pass (as long as the syntax is valid). But the changes will never be committed.

Instead, you should run `--fix` in local (can use pre-commit to help contributors), not in CI.
Result: if there's a discrepancy (contributor didn't run the fix), the CI will successfully reject it.